### PR TITLE
Backend integration test

### PR DESCRIPTION
**Pull request by baurine**
What did:

- Add backend integration test for user module

### CODE REVIEW

File: pkg/apiserver/user/auth.go [hunk 1]
```diff
@@ -223,10 +223,10 @@ func (s *AuthService) authForm(f AuthenticateForm) (*utils.SessionUser, error) {
 	return u, nil
 }
 
-func registerRouter(r *gin.RouterGroup, s *AuthService) {
```
Comment (baurine) [inline pkg/apiserver/user/auth.go:226]: TODO: revert

File: pkg/tidb/proxy.go [hunk 2]
```diff
@@ -148,12 +149,12 @@ func (p *proxy) pickActiveConn() (out net.Conn) {
 // pick returns an active remote if there is any.
 func (p *proxy) pick() *remote {
 	var picked *remote
-	if p.current == "" {
```
Comment (baurine) [inline pkg/tidb/proxy.go:151]: It is reported that it exists the data race between L#151 and L$156 when testing.

File: tests/integration/user/user_test.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+	var jsonByte []byte
+	if param != nil {
+		jsonByte, _ = json.Marshal(param)
+	}
+
+	w := httptest.NewRecorder()
+	c, _ := gin.CreateTestContext(w)
+	c.Request, _ = http.NewRequest(method, uri, bytes.NewReader(jsonByte))
+	return c, w
+}
```
Comment (shhdgit) [inline tests/integration/user/user_test.go:87]: Maybe we can just use `CreateTestContext` to test. Then most cases we don't need to validate *httptest.ResponseRecorder.

gin_test_helper.go
```
func TestReqWithHandlers(req *http.Request, handlers ...gin.HandlerFunc) (*gin.Context, *httptest.ResponseRecorder) {
	w := httptest.NewRecorder()
	c, e := gin.CreateTestContext(w)
	c.Request = req
	e.Any(req.RequestURI, handlers...)
	e.HandleContext(c)
	return c, w
}
```
xxx_test.go
```
req, _ := http.NewRequest("POST", "/user/login", bytes.NewReader(jsonByte))
ctx, _ := TestReqWithHandlers(req, s.FeatureTopSQL.VersionGuard(), provideTestDB, s.authService.LoginHandler)

...
// s.Require().True(errorx.IsOfType(ctx.Errors.Last().Err, rest.ErrBadRequest))
// s.Require().Equal(401, ctx.Writer.Status())
```

Comment (baurine) [reply tests/integration/user/user_test.go:87]: Let me have a look

Comment (baurine) [reply tests/integration/user/user_test.go:98]: it works, and `TestReqWithHandlers` can be refined to:

```go
	w := httptest.NewRecorder()
	c, e := gin.CreateTestContext(w)
	c.Request = req
	e.Handle(req.Method, req.URL.Path, handlers...)
	e.HandleContext(c)
	return c, w
```

`req.RequestURI` is empty.

**Comment by breeswish** [comment]
It is ok to review this PR? cc @YiniXu9506 @shhdgit

**Comment by baurine** [comment]
> It is ok to review this PR? cc @YiniXu9506 @shhdgit

Yep, it is ready to review.

**Comment by YiniXu9506** [comment]
> It is ok to review this PR? cc @YiniXu9506 @shhdgit

Sure！

**Review by YiniXu9506 — APPROVED** [review]
lgtm

**Review by shhdgit — APPROVED** [review]
LGTM

**Comment by breeswish** [comment]
please fix lints.

**Pull request merged**